### ISSUE TITLE
Team 1 return comments

### ISSUE THREAD

**Issue by dillam** [GatorGauge/GatorGauge#16]
The first step for our portion of the project is to analyze the comments of java source code. The first steps of this should be to create a python program to return the commented text and the number of comments. The program should be able to differentiate between single and multi-line comments. I have created a branch for this issue called team1/read_comments.

**Comment by shafferz** [comment]
The lab specification recommends we use [Dr. Kapfhammer's gatorgrader](https://github.com/gkapfham/gatorgrader), which was also developed in Python. 

I also think that [this Python module](https://pypi.python.org/pypi/comment_parser/1.0.3) for parsing comments that supports multiple code types may be useful to us, if we wanted to take a from-scratch approach. 

Thoughts?

**Comment by mariakimheinert** [comment]
@shafferz are you investigating the module or should I? I would really like to be able to contribute for tomorrow.

**Comment by shafferz** [comment]
@yeeunmariakim You can go ahead and investigate the module, that will free me up to draw up some ideas I've been thinking about since class today. I'm also thinking of investigating a way to parse through the parts of the JavaDoc comments specifically, so let me know what you find!

**Comment by mariakimheinert** [comment]
@dillam @shafferz I looked into the `comment_parser` module, but it appears that there is a bug in the dependency. Instead, I opted to use Professor Kapfhammer's code from GatorGrader, with a few modifications such as a new regex expression and functions to actually return lists of comments. This code is contained in the `feature/i-16-parse-comments` branch.

**Comment by delgrecoj** [comment]
@yeeunmariakim several testcases for the comment parsing functionality exist on my branch, `feature/i-16-parsing-comments`, which I'll further develop (the multi-line cases, in particular) before we eventually merge and adapt them to your parsing code.

**Comment by delgrecoj** [comment]
Pull request #43 has working comment extraction and testcases; waiting on merge into master.

**Comment by mariakimheinert** [comment]
Closing because of merge.
